File: examples/python/streaming.py
"""
AIP Streaming — Subscribe to a streaming intent (SSE).

Usage:
    export AIP_API_KEY="sk-your-key"
    python streaming.py
"""
import os
from agent_intent_protocol import AIPClient


def main():
    client = AIPClient(api_key=os.environ.get("AIP_API_KEY"))

    print("─── Streaming chat completion ───")
    stream = client.subscribe(
        intent="chat_completion",
        payload={
            "messages": [
                {"role": "user", "content": "Write a haiku about distributed systems."}
            ],
            "stream": True,
        },
        optimize_for="speed",
    )

    print("  Response: ", end="", flush=True)
    for chunk in stream:
        if chunk.delta:
            print(chunk.delta, end="", flush=True)
    print(f"\n  Total tokens: {stream.usage.total_tokens}")
    print(f"  Cost: ${stream.cost_usd}")


if __name__ == "__main__":
    main()

File: examples/typescript/streaming.ts
import { AIPClient } from '@jarvisclaw/agent-intent-protocol';

/**
 * AIP Streaming — SSE stream for chat completion.
 *
 * Usage:
 *   export AIP_API_KEY="sk-your-key"
 *   npx tsx streaming.ts
 */
async function main() {
  const client = new AIPClient({ apiKey: process.env.AIP_API_KEY });

  console.log('─── Streaming chat completion ───');
  const stream = await client.subscribe({
    intent: 'chat_completion',
    payload: {
      messages: [{ role: 'user', content: 'Write a haiku about distributed systems.' }],
      stream: true,
    },
    preferences: { optimizeFor: 'speed' },
  });

  process.stdout.write('  Response: ');
  for await (const chunk of stream) {
    if (chunk.delta) {
      process.stdout.write(chunk.delta);
    }
  }
  console.log(`\n  Total tokens: ${stream.usage.totalTokens}`);
  console.log(`  Cost: $${stream.costUsd}`);
}

main().catch(console.error);
